text_generation/causal_lm/cpp/continuous_batching/library/src/sequence_group.hpp

```diff
@@ -239,7 +245,13 @@ class SequenceGroup {
     void preempt_tokens(size_t num_preempt_tokens) {
         OPENVINO_ASSERT(num_preempt_tokens <= m_num_processed_tokens);
         m_num_processed_tokens -= num_preempt_tokens;
-        // Note, that m_max_content_len is kept as is
+        m_max_content_len -= num_preempt_tokens;
+
+        // this removal of tokens prevents duplicating of generated tokens after preemption of a sequence
+        // TODO: need to remove this when sampling is fixed to properly handle the case when sequnce group is returned after preemption
+        for (auto seq: m_sequences) {
+            seq->remove_tokens(std::min<size_t>(num_preempt_tokens, seq->get_generated_len()));
```

Comment:
could you please try to fix `requires_sampling` method implementation to return `true` only when we can generate really new tokens. 
E.g. because preemption we can compute only KV cache, but new tokens are not generated.

I hope that we require sampling when m_context_len >= max_context_len (or generated_ids.size()) ?